File: app/src/main/java/cn/guluwa/gulumusic/utils/ImageViewAttrAdapter.java
package cn.guluwa.gulumusic.utils;

import android.databinding.BindingAdapter;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.PixelFormat;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.support.annotation.NonNull;
import android.support.v7.graphics.Palette;
import android.view.View;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;

import okhttp3.internal.Util;

/**
 * Created by guluwa on 2018/1/12.
 */

public class ImageViewAttrAdapter {

    @BindingAdapter("android:src")
    public static void setSrc(ImageView view, Bitmap bitmap) {
        view.setImageBitmap(bitmap);
    }

    @BindingAdapter("android:src")
    public static void setSrc(ImageView view, int resId) {
        view.setImageResource(resId);
    }

    @BindingAdapter({"circleImageUrl"})
    public static void loadCircleImage(ImageView imageView, String url) {
        Glide.with(imageView.getContext())
                .load(url)
                .apply(new RequestOptions().override(AppUtils.dp2px(imageView.getContext(), 50)).circleCrop())
                .into(imageView);
    }

    @BindingAdapter({"bigImageUrl"})
    public static void loadBigImage(ImageView imageView, String url) {
        Glide.with(imageView.getContext())
                .load(url)
                .into(imageView);
    }
}

File: app/src/main/java/cn/guluwa/gulumusic/utils/ImageViewAttrAdapter.kt
package cn.guluwa.gulumusic.utils

import android.databinding.BindingAdapter
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.graphics.Canvas
import android.graphics.PixelFormat
import android.graphics.drawable.BitmapDrawable
import android.graphics.drawable.Drawable
import android.support.v7.graphics.Palette
import android.view.View
import android.widget.ImageView

import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions

import okhttp3.internal.Util

/**
 * Created by guluwa on 2018/1/12.
 */


@BindingAdapter("android:src")
fun ImageView.setSrc(bitmap: Bitmap) {
    this.setImageBitmap(bitmap)
}

@BindingAdapter("android:src")
fun ImageView.setSrc(resId: Int) {
    this.setImageResource(resId)
}

@BindingAdapter("circleImageUrl")
fun ImageView.loadCircleImage(url: String?) {
    Glide.with(context)
            .load(url)
            .apply(RequestOptions().circleCrop())
            .into(this)
}

@BindingAdapter("imageUrl")
fun ImageView.loadImage(url: String?) {
    Glide.with(context)
            .load(url)
            .apply(RequestOptions().centerCrop())
            .into(this)
}

@BindingAdapter("bigImageUrl")
fun ImageView.loadBigImage(url: String?) {
    Glide.with(context)
            .load(url)
            .into(this)
}
